workers.tf

```diff
@@ -76,7 +76,7 @@ resource "outscale_vm" "workers" {
     connection {
       type                = "ssh"
       user                = "outscale"
-      host                = var.public_cloud ? outscale_public_ip.workers[count.index].public_ip : format("10.0.1.%d", 19 + count.index)
+      host                = var.public_cloud ? self.public_ip : format("10.0.1.%d", 19 + count.index)
```

Comment:
You use this self ?
https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax